flume-ng-core/src/main/java/org/apache/flume/SinkRunner.java

```diff
@@ -166,6 +166,13 @@ public void run() {
           } catch (InterruptedException ex) {
             Thread.currentThread().interrupt();
           }
+        } catch (Throwable th){
```

Comment:
In general, catching `Error` is to be avoided: https://docs.oracle.com/javase/8/docs/api/java/lang/Error.html . Catching the superclass `Throwable` falls under the same category.

Can you think of any alternative ways of improvement?

Reply:
@lightzhao, OutOfMemoryError means Flume has run out of memory. Can you increase the amount of memory you give to Flume instead?

You also may want to consider more carefully tuning the byteCapacity of your memory channel. See https://flume.apache.org/FlumeUserGuide.html#memory-channel

Either way, I don't think this patch is a good way to work around your OOME problem.